resources/packs/gcp/common.go

```diff
@@ -54,3 +54,44 @@ func getAssetIdentifier(runtime *resources.Runtime) *assetIdentifier {
 	}
 	return &assetIdentifier{name: name, region: region, project: project}
 }
+
+type resourceId struct {
+	Project string
+	Region  string
+	Name    string
+}
+
+func getNetworkByUrl(networkUrl string, runtime *resources.Runtime) (interface{}, error) {
+	// A reference to a network is not mandatory for this resource
+	if networkUrl == "" {
+		return nil, nil
+	}
+
+	// Format is https://www.googleapis.com/compute/v1/projects/mondoo-edge/global/networks/mondoo-gke-cluster-2-net
+	params := strings.TrimPrefix(networkUrl, "https://www.googleapis.com/compute/v1/")
+	parts := strings.Split(params, "/")
+	resId := resourceId{Project: parts[1], Region: parts[2], Name: parts[4]}
+
+	return runtime.CreateResource("gcp.project.computeService.network",
+		"name", resId.Name,
+		"projectId", resId.Project,
+	)
+}
+
+func getSubnetworkByUrl(subnetUrl string, runtime *resources.Runtime) (interface{}, error) {
+	// A reference to a subnetwork is not mandatory for this resource
+	if subnetUrl == "" {
+		return nil, nil
+	}
+
+	// Format is https://www.googleapis.com/compute/v1/projects/mondoo-edge/regions/us-central1/subnetworks/mondoo-gke-cluster-2-subnet
```

Comment:
I'd not include links to internal resources in public code. It's an authorized endpoint, but it is already leaking information.

The data is helpful as a comment, perhaps we can include it directly and change some of the internal names.

Reply:
solved